services/report/labels_index.py

```diff
@@ -2,45 +2,68 @@
 
 from shared.reports.resources import Report
 
+from database.models.core import Repository
 from database.models.reports import CommitReport
 from services.archive import ArchiveService
+from services.report.report_builder import SpecialLabelsEnum
 
 
 class LabelsIndexService(object):
     _archive_client: ArchiveService
+    commit_report_code: str
+    commit_sha: str
 
-    def __init__(self, commit_report: CommitReport) -> None:
-        self.commit_report = commit_report
```

Comment:
Curious about this change - what's wrong with passing in the commit report?

Reply:
There's nothing wrong with doing that per se, but it doesn't match all usages.

In the carry forward bit the `commit_report` doesn't exist yet - or I don't have access to it (for the parent commit) - and yet I create `LabelIndexServices` for the commits. So I needed to be able to create them without the `commit_report`.

Reply:
I have since taken Matt's suggestion around that bit of code tho, so I can revert back the code if preferred.